### ISSUE TITLE
Offline mode: could not Download EGAPx related files from NCBI

### ISSUE THREAD

**Issue by bismarck1008** [ncbi/egapx#36]
python3 ui/egapx.py -dl -lc ../EGAPx_local

!!WARNING!!
This is an alpha release with limited features and organism scope to collect initial feedback on execution. Outputs are not yet complete and not intended for production use.

Downloading gnomon/2
Traceback (most recent call last):
  File "/data/bio-software/egapx/ui/egapx.py", line 810, in <module>
    sys.exit(main(sys.argv))
             ^^^^^^^^^^^^^^
  File "/data/bio-software/egapx/ui/egapx.py", line 683, in main
    download_egapx_ftp_data(args.local_cache)
  File "/data/bio-software/egapx/ui/egapx.py", line 178, in download_egapx_ftp_data
    ftpd.connect(FTP_EGAP_SERVER)
  File "/data/bio-software/egapx/ui/egapx.py", line 67, in connect
    self.reconnect()
    ^^^^^^^^^^^^^^^^
  File "/data/bio-software/egapx/ui/egapx.py", line 71, in reconnect
    self.ftp.login()
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/ftplib.py", line 414, in login
    resp = self.sendcmd('PASS ' + passwd)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/ftplib.py", line 281, in sendcmd
    return self.getresp()
           ^^^^^^^^^^^^^^
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/ftplib.py", line 244, in getresp
    resp = self.getmultiline()
           ^^^^^^^^^^^^^^^^^^^
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/ftplib.py", line 230, in getmultiline
    line = self.getline()
           ^^^^^^^^^^^^^^
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/ftplib.py", line 212, in getline
    line = self.file.readline(self.maxline + 1)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/bio-software/mambaforge/envs/egapx/lib/python3.12/socket.py", line 720, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
TimeoutError: [Errno 110] Connection timed out

**Comment by boukn** [comment]
it looks like it might be network access?  Does this command work?

`
$ curl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/EGAP/support_data/target_proteins/2/taxid.list`

**Comment by bismarck1008** [comment]
The link you gave can be downloaded, but I checked the above ftp site and found that the network is not accessible, there may be restrictions. I'll think of another way. Thank you very much!

> it looks like it might be network access? Does this command work?
> 
> ` $ curl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/EGAP/support_data/target_proteins/2/taxid.list`

The link you gave can be downloaded, but I checked the above ftp site and found that the network is not accessible, there may be restrictions. I'll think of another way. Thank you very much!
